python_modules/dagster-pipes/dagster_pipes/__init__.py

```diff
@@ -721,6 +726,152 @@ def load_context(self, params: PipesParams) -> Iterator[PipesContextData]:
             )
 
 
+class ExcThread(threading.Thread):
+    def run(self, *args, **kwargs):
+        self.exceptions = queue.Queue()
+
+        try:
+            super().run(*args, **kwargs)
+        except:
+            self.exceptions.put(sys.exc_info())
+
+    def join(self, *args, **kwargs):
+        super().join(*args, **kwargs)
+
+        while not self.exceptions.empty():
+            exc_info = self.exceptions.get()
+            sys.stderr.write(traceback.format_exception(*exc_info[1].with_traceback(exc_info[2])))
+
+
+class PipesStdioLogWriter(
+    PipesLogWriter[T_LogChannel]
+):  # log writers can potentially capture other type sof logs (for example, from Spark workers)
+    """Log writer which collects stdout and stderr of the current process should inherit from this class."""
+
+    LOGS_DIR_KEY = "logs_dir"
+
+    @abstractmethod
+    def make_channel(
+        self, params: PipesParams, stream: Literal["stdout", "stderr"]
+    ) -> T_LogChannel:
+        pass
+
+    @contextmanager
+    def open(self, params: PipesParams) -> Iterator[None]:
+        is_session_closed = threading.Event()
+        stdout_capturing_started = threading.Event()
+        stderr_capturing_started = threading.Event()
+
+        with ExitStack() as stack:
+            try:
+                stdout_channel = self.make_channel(params, stream="stdout")
+                stderr_channel = self.make_channel(params, stream="stderr")
+
+                stack.enter_context(
+                    stdout_channel.capture(stdout_capturing_started, is_session_closed)
+                )
+                stack.enter_context(
+                    stderr_channel.capture(stderr_capturing_started, is_session_closed)
+                )
+
+                # make sure capturing is working
+                stdout_capturing_started.wait()
+                stderr_capturing_started.wait()
+
+                time.sleep(0.1)  # make sure it's REALLY started
+
+                yield
+            finally:
+                is_session_closed.set()
+
+
+class PipesStdioLogWriterChannel(PipesLogWriterChannel):
+    """A base class for log writer channels that capture stdout and stderr of the current process."""
+
+    def __init__(self, stream: Literal["stdout", "stderr"], interval: float, name: str):
```

Comment:
`name` has to be an argument because it's beneficial to include the target location for writing the logs in the name, but this class doesn't own it, it will only be known in child classes.